Failing test on null process group

```
ERROR] Failures (1):
[ERROR]   JUnit Jupiter:OperatingSystemTest:testProcessStrings()
[ERROR]     MethodSource [className = 'oshi.software.os.OperatingSystemTest', methodName = 'testProcessStrings', methodParameterTypes = '']
[ERROR]     => java.lang.AssertionError: Current running process group shouldn't be null
[ERROR] Expected: is not null
[ERROR]      but: was null
[ERROR]        org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
[ERROR]        oshi.software.os.OperatingSystemTest.testProcessStrings(OperatingSystemTest.java:99)
```
https://github.com/oshi/oshi/blob/3ce7589dfb0c56d76d7ca4e50291df8ac54cdda0/oshi-core/src/test/java/oshi/software/os/OperatingSystemTest.java#L99

This is on macOS, relevant code:
```java
            this.groupID = Integer.toString(taskAllInfo.pbsd.pbi_gid);
            Group grgid = SystemB.INSTANCE.getgrgid(taskAllInfo.pbsd.pbi_gid);
            if (grgid != null) {
                this.group = grgid.gr_name;
            }
```
Troubleshooting shows the pbi_gid resolves to a value but it gives an errno of 1 (EPERM) looking it up.

Test passes under sudo.

Problematic group appears to (probably) be one of wheel daemon kmem sys tty operator procview procmod  staff certusers  

Probably should default to a string representation of the GID on failure.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,6 @@
 # 6.4.3 (in progress)
 
-* Your contribution here!
+* [#2377](https://github.com/oshi/oshi/pull/2377): Graceful fallback for macOS Process user or group name - [@dbwiddis](https://github.com/dbwiddis).
 
 # 6.4.0 (2022-12-02), 6.4.1 (2023-03-18), 6.4.2 (2023-05-02)
 

--- a/oshi-core/src/main/java/oshi/software/os/mac/MacOSProcess.java
+++ b/oshi-core/src/main/java/oshi/software/os/mac/MacOSProcess.java
@@ -441,14 +441,10 @@ public class MacOSProcess extends AbstractOSProcess {
             this.parentProcessID = taskAllInfo.pbsd.pbi_ppid;
             this.userID = Integer.toString(taskAllInfo.pbsd.pbi_uid);
             Passwd pwuid = SystemB.INSTANCE.getpwuid(taskAllInfo.pbsd.pbi_uid);
-            if (pwuid != null) {
-                this.user = pwuid.pw_name;
-            }
+            this.user = pwuid == null ? Integer.toString(taskAllInfo.pbsd.pbi_uid) : pwuid.pw_name;
             this.groupID = Integer.toString(taskAllInfo.pbsd.pbi_gid);
             Group grgid = SystemB.INSTANCE.getgrgid(taskAllInfo.pbsd.pbi_gid);
-            if (grgid != null) {
-                this.group = grgid.gr_name;
-            }
+            this.group = grgid == null ? Integer.toString(taskAllInfo.pbsd.pbi_gid) : grgid.gr_name;
             this.threadCount = taskAllInfo.ptinfo.pti_threadnum;
             this.priority = taskAllInfo.ptinfo.pti_priority;
             this.virtualSize = taskAllInfo.ptinfo.pti_virtual_size;

